EIPS/eip-2027.md

```diff
@@ -0,0 +1,74 @@
+---
+eip: 2027
+title: State Rent C - Net contract size accounting
+author: Alexey Akhunov (@AlexeyAkhunov)
+discussions-to: https://ethereum-magicians.org/t/eip-2027-net-contract-size-accounting-change-c-from-state-rent-v3-proposal/3275
+status: Draft
+type: Standards Track
+category: Core
+created: 2019-05-14
+---
+
+<!--You can leave these HTML comments in your merged EIP and delete the visible duplicate text guides, they will not appear and may be helpful to refer to if you edit it again. This is the suggested template for new EIPs. Note that an EIP number will be assigned by an editor. When opening a pull request to submit your EIP, please use an abbreviated title in the filename, `eip-draft_title_abbrev.md`. The title should be 44 characters or less.-->
+
+## Simple Summary
+<!--"If you can't explain it simply, you don't understand it well enough." Provide a simplified and layman-accessible explanation of the EIP.-->
+Ethereum starts counting the number of storage slots filled and emptied in the contracts. Since the number of pre-existing slots is not currently accounted
+in the state, effectively, only net change in the number of slots is tracked. In the subsequent change, called *Gross contract size accounting*, the total
+number of storage slots starts being tracked.
+
+## Abstract
+<!--A short (~200 word) description of the technical issue being addressed.-->
+It is part of the State Rent roadmap. This particular change introduced initial, net accounting of the number of the contract storage slots. Though not very
+useful on its own, it makes it possible to introduce gross accounting of the number of storage slots, which is useful for number of things:
+1. Gas cost of operations suchs as `SLOAD` and `SSTORE` will need to be increased to compensate for extra bandwidth consumed by the block proofs. Thought in
+the beginning, the cost would be fixed, it later be automatically calibrated depending on the size of the contract `SLOAD` and `SSTORE` operate on.
```

Comment:
Pardon my ignorance, but I was under the impression that these costs were getting larger because the entire state (of all contracts) increases, not just for a single contract. Is there a good place where I can read about this? This implies that `SLOAD`s  / `SSTORE`s on small contracts are less computationally expensive than on large contracts (small/large meaning the storage size of the contract here).

Reply:
Yes, `SLOAD` and `SSTORE` do get more computationally expensive on large contracts (at least in the most currently implementations). They also get more expensive when the entire state database grows. I tried to describe this in my blog post series, but, of course, I left a lot of things implicit, otherwise I would need to write a while book on it: https://medium.com/@akhounov/looking-back-at-the-ethereum-1x-workshop-26-28-01-2019-part-2-d3d8fdcede10

Reply:
Great write up. thanks!

Reply:
> Thought
should be `though`
> it later
word missing

Reply:
Thank you, fixed